internal/sharecode/sharecodedata/code.go

```diff
@@ -0,0 +1,59 @@
+package sharecodedata
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"log/slog"
+	"net/url"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/random"
+)
+
+func Generate() (PlainText, Hashed) {
```

Comment:
this is the main change, replacing `random.String(12)` so you get a plaintext and hashed version that are then only used in the appropriate places